dashboard/app/jobs/aichat_request_chat_completion_job.rb

```diff
@@ -0,0 +1,80 @@
+class AichatRequestChatCompletionJob < ApplicationJob
+  queue_as :default
+
+  before_enqueue do |job|
+    request = job.arguments.first[:request]
+    request.update(execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:QUEUED])
+  end
+
+  before_perform do |job|
+    request = job.arguments.first[:request]
+    request.update(execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:RUNNING])
+  end
+
+  # Catch any exceptions that occur during the job and update the request status accordingly
+  rescue_from Exception do |exception|
+    if rack_env?(:development)
+      puts "AichatRequestChatCompletionJob Error: #{exception.full_message}"
+    end
+
+    request = arguments.first[:request]
+    request.update(response: exception.message, execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE])
```

Comment:
unless you have a specific reason, I would recommend using `update!` instead of `update`, so that a failure to update is never silently ignored. here and below